extensions/oidc-client/deployment/src/test/java/io/quarkus/oidc/client/NamedOidcClientInjectionTestCase.java

```diff
@@ -0,0 +1,56 @@
+package io.quarkus.oidc.client;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.restassured.RestAssured;
+
+@QuarkusTestResource(KeycloakRealmUserPasswordManager.class)
+public class NamedOidcClientInjectionTestCase {
```

Comment:
@fwippe I believe it would be good to check the token content - as it is not guaranteed here the right client is invoked.
You should update the Keycloak test resource to create 2 more realms per each extra named client (and update the named clients in the config to point to those realms)- and in the test, when you get the token, check the `issuer` - it would be realm specific. `OidcUtils` has a simple utility to return a `JsonObject` representing the JWT claims.

Reply:
Makes sense, thanks for your input! I will work on improving the test.

Reply:
Introduced a new `Keycloak` resource helper class `KeycloakMultiRealmsUserPasswordManager` to create two realms named "realm1" and "realm2". Issuer of resulting tokens is validated accordingly.

Reply:
@fwippe - thanks, I'd propose to modify the existing `KeycloakRealmUserPasswordManager` instead, sorry I should've clarified it earlier: your new test manager code is perfect - but we are having too many of these test resources duplicating each other (hence the reason behind introducing `test-security-keycloak` with a few integration tests already updated to reuse a KC test manager coming from that module) - so it will be just simpler for us later on to keep removing these extra test support code - one less resource to handle.
`KeycloakRealmUserPasswordManager` already has a `quarkus` realm - so please just add a `quarkus2` realm - and that should be enough for now

Reply:
@sberyozkin ok, removed dedicated multi realm Keycloak resource and instead added additional realm `quarkus2` in `KeycloakRealmUserPasswordManager`. had to reorder all other realms so that I didn't have to include realms `quarkus` and `quarkus7` but instead `quarkus` and `quarkus2`.

Reply:
@fwippe - I'm sorry, that was not a good naming scheme I used earlier.

Sorry for being a pain - here is an alternative idea:
- keep the single `quarkus` realm, but [here](https://github.com/quarkusio/quarkus/blob/main/extensions/oidc-client/deployment/src/test/java/io/quarkus/oidc/client/KeycloakRealmUserPasswordManager.java#L33) add one more user, `("bob", user)`. And then check not `iss` but `preferred_username` - that would let you revert those other test resource updates
Thanks